### PR TITLE
feat(41): add --version to all CLI bins

### LINKED ISSUE #41: add --version

### PR DESCRIPTION
**Pull request by ralphkrauss**
## Summary

- Adds `--version` (and `--version --json`) to all four package bins: `agent-orchestrator`, `agent-orchestrator-daemon`, `agent-orchestrator-opencode`, `agent-orchestrator-claude`.
- Each bin prints `<bin-name> <version>\n` (or single-line `{ "name", "version" }` JSON), exits `0`, and short-circuits before any I/O — no daemon start, no wrapped-CLI spawn, no fs validation.
- Launcher bins (`-claude`, `-opencode`) intercept `--version` only when it is the **first own-arg** before the `--` passthrough boundary, so `<bin> -- --version` still forwards to the wrapped binary unchanged. A misplaced `--version` (e.g. `--print-config --version`) falls through to the parser and is rejected as an unknown option.
- Help text on each bin updated to list the new flag.

## Issue

Closes #41

## Implementation notes

- Shared `formatVersionOutput(binName, json)` helper in `src/packageMetadata.ts` keeps all four bins emitting identical shapes.
- `--version`/`--json` are intentionally not added to either launcher's known-flag set — keeping the launchers' top-level argv surface tight. The pre-parse own-args scan handles version detection.
- No new dependencies; reuses existing `getPackageMetadata()` which already reads the package's own `package.json`.

## Verification

- [x] `pnpm build` clean.
- [x] `pnpm test` — 546 tests, 544 pass, 2 skipped, 0 fail (includes 14 new tests in `cliVersion.test.ts`).
- [x] All four bins spawned with `--version` and `--version --json` and asserted to print the expected output.
- [x] Subcommand routing tested: `agent-orchestrator claude --version` and `agent-orchestrator opencode --version` print the orchestrator version.
- [x] Regression: misplaced `--version` (e.g. `--print-config --version`) falls through and emits `Unknown option: --version`.
- [x] Passthrough verified: parser-level test confirms `-- --version` is forwarded to `claudeArgs` / `opencodeArgs` for downstream binaries.

Plan and review artifacts at `plans/41-add---version/`.

Generated with the help of Claude Code.

## Summary by CodeRabbit

* **New Features**
  * Added a top-level --version (and --version --json) to all CLI binaries to print package name/version; help text updated.

* **Bug Fixes**
  * Ensured version handling exits early and does not start daemons or spawn tools; launcher argument parsing adjusted to avoid misinterpreting misplaced --version.

* **Tests**
  * Added end-to-end tests covering plain/JSON output, routing, passthrough, and edge-case placements of --version.

### CODE REVIEW

File: plans/41-add---version/plans/41-add-version-flag.md [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+| # | Candidate | Scope | Create After |
+|---|---|---|---|
+| 1 | None proposed | — | — |
+
+## Quality Gates
+
+- [x] `pnpm build` passes (`tsc` clean).
+- [x] `pnpm test` passes — 544 tests, 542 pass, 2 skipped, 0 fail (includes new `cliVersion.test.ts` with 12 tests).
+- [x] Manual smoke covered by spawn-based tests in `cliVersion.test.ts`.
```
[the remark this answers is not in the document]
Comment (ralphkrauss) [reply plans/41-add---version/plans/41-add-version-flag.md:111]: **[AI Agent]:** Fixed in commit 2c1d290. Quality Gates block now matches the Execution Log: 549 tests / 547 pass / 2 skipped / 0 fail, with 17 tests in `cliVersion.test.ts`.

**Comment by ralphkrauss** [comment]
**[AI Agent]:** Addressed CodeRabbit review nitpicks in commit 2c1d290:

- **`--version --json` coverage gap (cliVersion.test.ts:53-60)** — added three JSON-output assertions so every bin's `formatVersionOutput(..., true)` path is covered:
  - `daemon --version --json` (spawn-based)
  - `claude --version --json` (in-process via `runClaudeLauncher`)
  - `opencode --version --json` (in-process via `runOpenCodeLauncher`)
- **Split-import cleanup (cliVersion.test.ts:9-12)** — consolidated the four split imports into two `import { parser, run } from '<launcher>'` statements.

Test totals after the fix: `pnpm test` → 549 tests / 547 pass / 2 skipped / 0 fail; `cliVersion.test.ts` → 17 tests, all passing.
